### PR TITLE
use parent header passed from worker

### PR DESCRIPTION
**Pull request by madhur-tandon**

### CODE REVIEW

File: packages/kernel/src/kernel.ts [hunk 1]
```diff
@@ -228,13 +228,18 @@ export abstract class BaseKernel implements IKernel {
    * @param parentHeader The parent header.
    * @param content The stream content.
    */
-  protected stream(content: KernelMessage.IStreamMsg['content']): void {
+  protected stream(
+    content: KernelMessage.IStreamMsg['content'],
+    parentHeader: any = undefined
```
Comment (martinRenou) [inline packages/kernel/src/kernel.ts:233]:
```suggestion
    parentHeader: IHeader | {} | undefined = undefined
```

Comment (madhur-tandon) [reply packages/kernel/src/kernel.ts]: Done, Thanks!

**Comment by martinRenou** [comment]
Works like a charm 💯

**Comment by bollwyvl** [comment]
I just tried out this, works well:

```python
from ipywidgets import interact
from IPython.display import Markdown, display

@interact(cookies=slider)
def cookies(cookies=slider.value, calories=(0,150)):
    total_calories = calories * cookies
    if cookies:
        display(Markdown(f"If each cookie contains _{calories} calories_, _{cookies} cookies_ contain **{total_calories} calories**!"))
    else:
        display(Markdown(f"No cookies!"))
    if total_calories > 2000:
        display(Markdown(f"> Maybe that's too many cookies..."))
```

Given this can be a tricky one, but also a super useful example, may let's add that to the end of `pyolite - interactive-widgets.ipynb`?

**Comment by madhur-tandon** [comment]
@bollwyvl added the example in the latest commit. Thanks!

**Comment by martinRenou** [comment]
Great thanks!

**Pull request merged**